Don't add the first line of /etc/rpm/platform to the list of patterns
to match when computing an arch score, use it just for getting
information about the platform (cpu/vendor/os). Fixes #3864.

Upstream-Status: Inappropriate [embedded specific]

Index: rpm-5.4.14/lib/rpmrc.c
===================================================================
--- rpm-5.4.14.orig/lib/rpmrc.c
+++ rpm-5.4.14/lib/rpmrc.c
@@ -510,6 +510,7 @@ static rpmRC rpmPlatform(const char * pl
 #endif
 	}
 
+#if !defined(RPM_VENDOR_OE) /* Skip the explicit-platform */
 #if defined(RPM_VENDOR_OPENPKG) /* explicit-platform */
 	/* do not use vendor and GNU attribution */
 	p = rpmExpand("%{_host_cpu}-%{_host_os}", NULL);
@@ -520,7 +521,8 @@ static rpmRC rpmPlatform(const char * pl
 #endif
 	xx = mireAppend(RPMMIRE_STRCMP, 0, p, NULL, &mi_re, &mi_nre);
 	p = _free(p);
-	
+#endif
+
 	init_platform++;
     }
     rc = (init_platform ? RPMRC_OK : RPMRC_FAIL);
